Storage/SQL/SQLitePasswords.swift

```diff
@@ -67,8 +91,16 @@ private class PasswordsTable<T>: GenericTable<Password> {
 
     override var factory: ((row: SDRow) -> Password)? {
         return { row -> Password in
-            let site = Site(url: row["hostname"] as? String ?? "", title: "")
-            let pw = Password(site: site, username: row["username"] as? String ?? "", password: row["password"] as? String ?? "")
+            var user = row["username"] as? String ?? ""
+            var pass = row["password"] as? String ?? ""
+
+            let encType = EncryptionType(rawValue: row["encryptionType"] as? Int ?? EncryptionType.NONE.rawValue)
+            if encType == EncryptionType.AES256 {
+                let secret = SQLitePasswords.secretKey!
+                pass = pass.AES256DecryptWithKey(secret)! as String
+            }
+
+            let pw = Password(hostname: row["hostname"] as! String, username: user as String, password: pass)
```

Comment:
This is tricky. By reusing `pass` it is possible that encryption fails and that we store an unencrypted password. Better to use a new variable, encryptedPass, and use that. And not store antyhing in case of failures.